ut_core.py

```diff
@@ -0,0 +1,184 @@
+        group_output = self.session.write("s")
+        
+        # Extract group index from the output
+        group_index = self.find_index_in_output(group_output, group_name)
+        if group_index is None:
+            self.log.error(f"Group '{group_name}' not found in menu configuration.")
+            raise ValueError(f"Group '{group_name}' not found in the shell output.")
+        
+        self.log.info(f"Found group: {group_name}")
+        self.session.write(str(group_index))
+
+        
+        test_output = self.session.write("s")        
+        
+        # Extract test index from the output
+        test_index = self.find_index_in_output(test_output, test_name)
+        if test_index is None:
+            self.log.error(f"Test '{test_name}' not found in group '{group_name}'.")
+            raise ValueError(f"Test '{test_name}' not found in the shell output.")
+        
+        self.log.info(f"Found test: {test_name}")
+        self.session.write(str(test_index))
+
+
+    def find_index_in_output(self, output, target_name):
+        """
+        Finds the index of a target name in the shell output.
+
+        Args:
+            output (str): The shell output to search in.
+            target_name (str): The name of the group or test to find.
+
+        Returns:
+            int: The index of the target_name if found, otherwise None.
+        """
+        lines = output.splitlines()
+        for line in lines:
+            if target_name in line:
+                # Looking for the index e.g., "1. GroupName"
+                match = re.match(r"(\d+)\.", line.strip())
+                if match:
+                    return int(match.group(1))
+        return None
+
+
+    def collect_results(self, output):
+        """
+        Collects and interprets the results from the test execution output.
+
+        Args:
+            output (str): The output from the test execution.
+
+        Returns:
+            bool: True if the test passed successfully, False if the test failed, or None if the output format is unexpected.
+        """
+        
+        # Pattern to detect the number of suites, tests, and asserts with their corresponding results
+        # Run Summary:    Type  Total    Ran Passed Failed Inactive
+        #       suites      2      0    n/a      0        0
+        #        tests     16      1      1      0        0
+        #      asserts      2      2      2      0      n/a
+        run_summary_pattern = r"Run Summary:\s+Type\s+Total\s+Ran\s+Passed\s+Failed\s+Inactive"
+        summary_match = re.search(run_summary_pattern, output)
+
+        if summary_match:
+            # Extract the relevant lines for suites, tests, and asserts
+            suite_summary_line = re.search(r"suites\s+\d+\s+\d+\s+n/a\s+(\d+)\s+\d+", output)
+            test_summary_line = re.search(r"tests\s+\d+\s+\d+\s+(\d+)\s+(\d+)\s+\d+", output)
+            assert_summary_line = re.search(r"asserts\s+\d+\s+\d+\s+(\d+)\s+(\d+)\s+n/a", output)
```

Comment:
We might need to check with @Ulrond these patterns will consistently be in this order.